airflow/sensors/external_task.py

```diff
@@ -140,21 +140,30 @@ def __init__(
                 f"`{self.allowed_states}` and failed states `{self.failed_states}`"
             )
 
-        if external_task_id is not None and external_task_ids is not None:
+        if external_task_id and external_task_ids:
```

Comment:
Does `external_task_ids=[]` make sense? If it does, we’d want to keep using `is not None`.